tests/fixtures/__init__.py

```diff
@@ -72,18 +89,45 @@ def non_mocked_hosts() -> list[str]:
     return ["test"]
 
 
-@async_fixture(name="client")
-async def fixture_client(mongodb_fixture) -> AsyncGenerator[AsyncClient, None]:
-    """Get test client for the work package service"""
+async def publish_datasets(container: Container) -> None:
+    """Publish datasets to populate the database."""
+    config = container.config()
+    async with KafkaEventPublisher.construct(config=config) as event_publisher:
+        await event_publisher.publish(
+            payload=DATASET_OVERVIEW_EVENT.dict(),
+            type_="metadata_dataset_overview",
+            key="test_key",
+            topic="metadata",
+        )
+
+    event_subscriber = await container.event_subscriber()
+    await asyncio.wait_for(event_subscriber.run(forever=False), timeout=5)
+
+
+@async_fixture(name="container")
+async def fixture_container(
+    mongodb_fixture, kafka_fixture
```

Comment:
Makes it easier to work with:

```suggestion
    mongodb_fixture: MongoDbFixture, kafka_fixture: KafkaFixture
```

Reply:
Generally, we started to avoid populating state during the fixture setup. Exposing a method for setting state could make it more transparent and flexible see this as an example: https://github.com/ghga-de/hexkit/blob/main/hexkit/providers/s3/testutils.py#L91

Not something to address right now and can be discussed at a later point. Added to the refactoring list.

Reply:
The change was already made as part of the fix above.

Maybe we should have a "show house" service demonstrating all the best practices for testing.

Reply:
I like that idea. Added to the refactoring list.